portfolio/src/main/webapp/index.html

```diff
@@ -11,8 +11,8 @@
       <h1>Leonard Zhang's Portfolio</h1>
       <p>This is my portfolio. I just start modifying it and have not completed it yet. </p>
       <p>This is a photo of mine.</p>
-      <img src="/images/Leonard-3.jpg"/>
-      <p>Hello everyone, I'm Leonard Zhang. You can call me Leonard. I’m a rising junior at Carnegie Mellon University, 
+      <img class="first_picture" src="/images/Leonard-3.jpg"/>
```

Comment:
`first_picture` and especially `self-intro` sound like unique elements of this page to me. If so, using an `id` attribute instead of adding a class is probably better, unless you expect to reuse the rules across different pages in the site. So, if these are only going to appear on this page, I'd make these `id`s instead of `class`es, and then match them in your CSS with `#`. 

Also, my (outdated?) HTML/CSS knowledge says dashes are more common in IDs and class names compared to underscores, but either is probably fine; just be consistent within the project and the file.

Reply:
I will change the classes to ids. Also I just looked up on Google and it seems that dashes are more common that underscores in HTML/CSS. I will use dashes in the project.

To view the result of the rendered HTML, I think you can just clone my repo and went to the portfolio folder. After reaching the portfolio folder, run the command: mvn package appengine:run
Once you see the following line in the console:
INFO: Dev App Server is now running
you can click the third icon on the top left corner of Cloud shell and then select Preview on port 8080 to see the changed HTML page. 

Maybe the above instructions are not so clear. You can check this markdown file:
https://github.com/LeonardZ0731/step/blob/master/walkthroughs/week-2-web-development/portfolio-walkthrough.md#run-a-development-server
The section explains how to run the server in detail.

Reply:
Thanks! I was able to get this working on Crostini, as well. (Didn't try Cloud Shell, but I'm sure that was is easier.)